on_startup.py

```diff
@@ -135,5 +144,28 @@ def main():
 
 
 if __name__ == '__main__':
-    main()
+    try:
+        main()
+    except Exception as e:
```

Comment:
Failing test seems to be due to the fact  that we're catching something with this expect that we weren't before

I remember there being something off with the `django.setup()` within main. Maybe try pulling that out as a call before main and see what happens?